### PR TITLE
Adds EventTracker and Purchase events

### PR DESCRIPTION
**Pull request by wyattjoe40**
* Adds an `ATTNEventTracker`. This class is a singleton so it's easily accessible anywhere in the app. This class is used to send events to Attentive.
* Adds OCMock library for testing
* Updates Example app with a button to send a Purchase

### CODE REVIEW

File: Tests/ATTNSDKTest.m [hunk 1]
```diff
@@ -33,7 +33,7 @@ - (void)tearDown {
 - (void)testExample {
     ATTNUserIdentity* identity = [[ATTNUserIdentity alloc] init];
     ATTNAPI* api = [[ATTNAPI alloc] init];
-    ATTNSDK* sdk = [[ATTNSDK alloc] initWithDomain:@"domainValue" mode:@"Production" userIdentity:identity api:api];
+    // ATTNSDK* sdk = [[ATTNSDK alloc] initWithDomain:@"domainValue" mode:@"Production" userIdentity:identity api:api];
```
Comment (Mackey22) [inline Tests/ATTNSDKTest.m:36]: should we remove this file altogether?

**Review by Mackey22 — COMMENTED** [review]
Lgtm. what are the ramifications of adding OCMock? Any build issues that could come up?

**Comment by wyattjoe40** [comment]
> Lgtm. what are the ramifications of adding OCMock? Any build issues that could come up?

I added OCMock to just the Unit Test build/product, not the actual SDK library, so I don't think there are many ramifications, and if they are they should be isolated to the Unit Test build/product.

File: Tests/ATTNSDKTest.m (hunk 1 above)
Comment (wyattjoe40) [reply Tests/ATTNSDKTest.m]: yeah good call, removed

**Review by Mackey22 — APPROVED** [review]

**Pull request merged**